### PR TITLE
feat: payment sequence flow

### PR DESCRIPTION
**Pull request by andrehadianto**
[Notion link](https://www.notion.so/Sequence-of-flow-for-Payment-collected-dialog-box-and-error-dialog-box-6702ba3744e84ab3b99153e5914326cc) 

This PR adds... 
- separate checkoutConfirmation and add PENDING_CONFIRMATION state
- add useEffect in `ItemsSelectionCard` to check whether cartState is `PENDING_CONFIRMATION`

- [x] I've kept this PR as small as possible (~600 lines) by splitting it into PRs with manageable chunks of code
- [x] I've requested reviews from 1-2 reviewers
- [ ] I've added [accessibility IDs](https://www.notion.so/dc5f4ce910f7431c84344ac79344e9f5?v=d487366816834dd4ab8dc12e0b5928f3) to my components
- [ ] I've added/updated unit/integration tests
- [ ] I've added/updated [e2e tests](https://www.notion.so/e2e-Documentation-a096d3e0bf75485e85ad692af8371ef3) for at least the happy flow
- [ ] I've added [Chinese translations for i18n setup](https://www.notion.so/Translations-Dev-Guide-8c1da838982a4f59a386ec96ac6780c8)
- [x] I've added documentation in README/[Notion](https://www.notion.so/82b92fb1007640328dab9582c0a3694e?v=3b6ce48202cf40ad8450553799b13146)

### CODE REVIEW

File: src/hooks/useCart/useCart.tsx [hunk 1]
```diff
@@ -222,35 +230,15 @@ export const useCart = (
   );
 
   /**
-   * Handles the checking out of the cart.
    * Sets checkoutResult to the response of the post transaction.
```
Comment (sebastianquek) [inline src/hooks/useCart/useCart.tsx:233]: Please add a comment on the purpose of this function too!

File: src/hooks/useCart/useCart.tsx [hunk 2]
```diff
@@ -288,18 +276,61 @@ export const useCart = (
         }
       }
     };
+    complete();
+  }, [cart, ids, selectedIdType, authKey, endpoint, features?.apiVersion]);
+
+  /**
+   * Handles the checking out of the cart.
```
Comment (sebastianquek) [inline src/hooks/useCart/useCart.tsx:283]: Will be good to add that if there's a need to confirm payment, then the other `completeCheckout` must be called

File: src/hooks/useCart/useCart.test.tsx [hunk 3]
```diff
@@ -595,6 +595,67 @@ describe("useCart", () => {
       expect(result.current.cartState).toBe("UNSUCCESSFUL");
     });
 
+    it("should show PENDING_CONFIRMATION cart state when a PAYMENT_RECEIPT identifier input exists", async () => {
```
Comment (sebastianquek) [inline src/hooks/useCart/useCart.test.tsx:598]: Would be good to have a test that calls the completeCheckout as well

File: src/hooks/useCart/useCart.test.tsx [hunk 4]
```diff
@@ -1203,6 +1203,147 @@ describe("useCart", () => {
...
+            },
+          ],
+          lastTransactionTime: transactionTime,
+          maxQuantity: 15,
+          quantity: 0,
+        },
+      ]);
+    });
+
+    it("should be able to complete a transaction without going through checkoutCart", async () => {
```
Comment (sebastianquek) [inline src/hooks/useCart/useCart.test.tsx:1277]: Hmm come to think of it ah, it'll be better if we restrict it such that the state must be "PENDING_CONFIRMATION" when `completeCheckout` is called. This ensures `checkoutCart`, with all its validations, have already been carried out

File: src/components/CustomerQuota/ItemsSelection/ItemsSelectionCard.tsx [hunk 5]
```diff
@@ -76,19 +103,6 @@ export const ItemsSelectionCard: FunctionComponent<ItemsSelectionCard> = ({
...
-          CONFIRMATION_MESSAGE.PAYMENT_COLLECTION,
-          checkoutCart
-        );
-      };
-    } else {
-      return checkoutCart;
-    }
-  };
-
   // TODO:
```
Comment (yongjiajie) [inline src/components/CustomerQuota/ItemsSelection/ItemsSelectionCard.tsx:106]: Was this check for `alert` moved somewhere else?

**Review by yongjiajie — CHANGES_REQUESTED** [review]

File: src/hooks/useCart/useCart.tsx [hunk 6]
```diff
@@ -222,35 +230,16 @@ export const useCart = (
...
-   * Sets checkoutResult to the response of the post transaction.
+   * POST the transaction and set the cartState according to completion
+   * or error caught.
    */
-  const checkoutCart: CartHook["checkoutCart"] = useCallback(() => {
-    const checkout = async (): Promise<void> => {
-      setCartState("CHECKING_OUT");
-
-      const allIdentifierInputs: IdentifierInput[] = [];
+  const closeCheckoutCart = useCallback(() => {
```
Comment (sebastianquek) [inline src/hooks/useCart/useCart.tsx:236]: Hmm "close" is a little strange in the context. Perhaps we can keep it simple with `_completeCheckout`? The underscore indicates it's an internal function

**Review by sebastianquek — APPROVED** [review]
LGTM!!

**Review by yongjiajie — APPROVED** [review]
LGTM

**Pull request merged**